include/net/net_config.h

```diff
@@ -44,6 +44,13 @@ extern "C" {
  */
 #define NET_CONFIG_NEED_IPV4   0x00000004
 
+/**
+ * @brief Application needs IP subsystem configured and initialized, and
+ * network connectivity established. The application does not care whether
+ * the connectivity is either IPv4 or IPv6 based.
+ */
+#define NEED_IP_CONNECTIVITY   0x00000008
```

Comment:
I just realized that the name of this define does not follow the other values, will fix it.